keras/engine/topology.py

```diff
@@ -465,6 +465,17 @@ def __call__(self, x, mask=None):
             x: can be a tensor or list/tuple of tensors.
             mask: tensor or list/tuple of tensors.
         '''
+        # place the op on the right device:
+        if type(x) in [list, tuple] and not hasattr(x[0], '_keras_on_device'):
```

Comment:
I don't really get the need for `run_on_device`. Surely this should be handled by the backend in the background, every time you instantiate an op / variable?